### ISSUE TITLE
PyCompileAndRuntime produces different results when running multiple times

### ISSUE THREAD

**Issue by axeabc** [onnx/onnx-mlir#2956]
Hi , I want to use `PyCompileAndRuntime`  to  CompileAndRun a onnx model (e.g. model.onnx)  and print the output of the model.onnx. However, this will produce different results when I repeatedly run the code.

`OnnxRuntime4Validate.py`
``` 
import pickle
from PyCompileAndRuntime import OMCompileExecutionSession
import json
import numpy as np
import os

def run_lib():

    session = OMCompileExecutionSession( 'model.onnx',"-O3",use_default_entry_point=False )
    # Query entry points in the model.
    entry_points = session.entry_points()
    for entry_point in entry_points:
        session.set_entry_point(name=entry_point)
        print(f'Run at {entry_point}')
        # print("input signature in json", session.input_signature())
        # print("output signature in json", session.output_signature())
        inputs = json.loads(session.input_signature())
        input_names = [item['name'] for item in inputs]
        data = {}
        with open('oracle.pkl', 'rb') as file:
            data = pickle.load(file)
        inputs = [data['input'][input_name] for input_name in input_names]
        outputs1 = session.run(inputs)
        print(outputs1)

if __name__ == "__main__":
    run_lib()
``` 
model.onnx and oralce.pkl  are  in  [model.zip](https://github.com/user-attachments/files/17150048/model.zip)

'python3 OnnxRuntime4Validate.py'
<img width="1144" alt="image" src="https://github.com/user-attachments/assets/813d0167-1a78-4d8e-a1b1-06bd8a2158ec">

**Comment by AlexandreEichenberger** [comment]
Would it make sense to also print the inputs, so that we are sure they are the same?

**Comment by axeabc** [comment]
> Would it make sense to also print the inputs, so that we are sure they are the same?

I cannot have a complete printscreen because the inputs are large.  The print results including inputs and outputs have been exported to two files: tmp1 and tmp2. The diff results shows that the outputs are different given the same inputs.
<img width="1115" alt="image" src="https://github.com/user-attachments/assets/5f669d78-df6e-48f3-8565-2ffdf857dc04">
The tmp1 and tmp2 can be found in 
[tmp.zip](https://github.com/user-attachments/files/17168213/tmp.zip)
The specific inputs and model can be found in 
[model.zip](https://github.com/user-attachments/files/17168217/model.zip)

**Comment by tungld** [comment]
Have you tried to compile the model with `-O0` to see if it works?

**Comment by axeabc** [comment]
> Have you tried to compile the model with `-O0` to see if it works?

Yes, the cases are the same (the results are different when running multiple times) from '-O0' to '-O2'
